### PR TITLE
🛠️ Fix ➾ memory leak in test/e2e/utils

### PR DESCRIPTION
**Pull request by mweichert**
# 🌮 Taqueria PR

## 🪁 Description

Memory leaks were found in the e2e/utils module.

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [x] ⛱️ I have completed this PR template in full and updated the title appropriately
- [x] ⛵ My code builds cleanly, and I have manually tested the changes
- [ ] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [x] 🏖️ New and existing unit tests pass locally and in CI
- [x] 🔱 The test plan has been implemented and verified by an SDET
- [x] 🦀 Automated tests have been written and added to this PR
- [x] 🐬 I have commented my code, particularly in hard-to-understand areas
- [x] 🤿 Corresponding changes have been made to all documentation
- [x] 🐚 Required changes to the VScE have been made
- [x] 🪸 Required updates to scaffolds have been made
- [ ] 🚢 The release checklist has been completed

## 🛩️ Summary of Changes

Found and resolved some issues with the utils module in the tests package.

## 🎢 Test Plan

Run `npm run test:e2e:debug` from the tests workspace for the following test suite:
```
import { generateTestProject } from './utils/utils';
describe("Test for memory leaks", () => {

  beforeAll(() => generateTestProject('test'))

  test("Simple test should succeed", () => {
    expect(true).toEqual(true)
  })
})
```

### CODE REVIEW

File: tests/e2e/utils/utils.ts [hunk 1]
```diff
@@ -132,19 +116,16 @@ export async function installDependencies(
...
+				encoding: 'utf8',
+			});
+		} else {
+			const output = await exec(`taq install @taqueria/plugin-${packageName}`, { cwd: `./${projectPath}` });
 		}
 
-		await checkFolderExistsWithTimeout(`./${projectPath}/node_modules/@taqueria/plugin-${packageName}/index.js`);
+		await checkFolderExistsWithTimeout(`${projectPath}/node_modules/@taqueria/plugin-${packageName}/index.js`);
 	}
+	return;
```
Comment (jchenche) [inline tests/e2e/utils/utils.ts:130]: Empty return?

File: tests/e2e/utils/utils.ts [hunk 2]
```diff
@@ -1,32 +1,24 @@
...
 
-export const sleep = (ms: number) => new Promise((resolve, reject) => setTimeout(resolve, ms));
+export const sleep = (ms: number) => new Promise((resolve, _reject) => setTimeout(resolve, ms));
 
 export const generateTestProject = async (
 	projectPath: string,
 	packageNames: string[] = [],
 	localPackages: boolean = true,
 ) => {
+	debugger;
```
Comment (jchenche) [inline tests/e2e/utils/utils.ts:15]: Remove?

File: tests/e2e/utils/utils.ts [hunk 3]
```diff
@@ -102,6 +85,7 @@ export async function waitFor<T>(fn: () => Promise<T>, opts = { timeout: 10000,
 		} catch (err) {
 			lastErr = err;
 			await sleep(interval);
+			elapsed += interval;
```
Comment (jchenche) [inline tests/e2e/utils/utils.ts:88]: Where is this elapsed coming from?

File: tests/e2e/utils/utils.ts (hunk 2 above)
Comment (mweichert) [reply tests/e2e/utils/utils.ts:15]: Good catch - done!

File: tests/e2e/utils/utils.ts (hunk 3 above)
Comment (mweichert) [reply tests/e2e/utils/utils.ts:88]: Here's the whole function:
```
export async function waitFor<T>(fn: () => Promise<T>, opts = { timeout: 10000, interval: 100 }) {
	let elapsed = 0;
	let lastErr = undefined;
	const timeout = opts.timeout ?? 1000;
	const interval = opts.interval ?? 100;

	while (elapsed < timeout) {
		try {
			return await fn();
		} catch (err) {
			lastErr = err;
			await sleep(interval);
			elapsed += interval;
		}
	}
	throw lastErr;
}
```

You can see that elapsed is defined on the first line of the function.

**Review by jchenche — APPROVED** [review]

**Review by AlirezaHaghshenas — APPROVED** [review]

**Comment by michaelkernaghan** [comment]
This can be closed. the leaking procudure is being removed

**Pull request closed without merging**